### PR TITLE
initial version of rocket stack

### LINKED ISSUE #393: Create a stack for rocket web framework
We need a stack for rocket web framework based on rust language.

### PR DESCRIPTION
**Pull request by skoh7645**
### Checklist:

- [x] Read the [Code of Conduct](https://github.com/appsody/website/blob/master/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md).

- [x] Followed the [commit message guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md#commit-message-guidelines).

- [x] Stack adheres to [Appsody stack structure](https://github.com/appsody/website/blob/master/content/docs/stacks/stack-structure.md).

### Contributing a new stack:
Initial version of rocket stack
- Describe how application dependencies are managed: Stack dependencies are included as a dependency in the template. Cargo caches crates registry which we are pointing to `APPSODY_DEPS`. The stack controls the rust and rocket version.

- Explain how Appsody file watcher is utilized: Only watches rust `.rs` files

- Describe other Appsody environment variables defined by the stack image: `CARGO_HOME` (location of cached crates) specified in rust docker image is overwritten as we can only mount one directory in `APPSODY_DEPS`. We only want to mount dependencies brought in by the user - but git dependencies are cached in a different directory from crate dependencies so needed to specify different directory to prevent mounting the cargo binaries in a volume

- Describe any limitations and known issues: 
- [ ] `appsody build` does not support multiple release binaries?
- [ ] Problems with caching stack dependencies locally (so running `cargo run` outside of `appsody run` will give an error saying it cannot find the stack dependencies)
- [ ] No debug support implemented

### Related Issues:
Fixes #393

### CODE REVIEW

File: experimental/rust-rocket/README.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+This stack is based on the `Rust v1.37` and `Rocket v0.4.2` and allows you to develop new or existing Rocket applications using Appsody.
+
+Currently, debugging is not supported in this stack.
+
+## Health checking
+
+Health-checking enables the cloud platform to determine the `liveness` of your application.
+
+ This exposed as follows:
```
Comment (Kamran64) [inline experimental/rust-rocket/README.md]: nit: This exposed as follows: -> This **is** exposed as follows

**Comment by Kamran64** [comment]
This looks ok to me - all commands are working properly and file structure looks fine. I think a second review from @neeraj-laad would be good though.

File: experimental/rust-rocket/README.md (hunk 1 above)
Comment (skoh7645) [reply experimental/rust-rocket/README.md]: Done

File: experimental/rust-rocket/image/project/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+FROM rust:1.37-slim
+
+WORKDIR "/project"
+
+# get files and built binary from previous image
+COPY --from=builder /project/user-app/run.sh /project/user-app/Cargo.toml /project/user-app/target/release/ ./
+
+RUN rm -rf build deps examples incremental
+
+RUN useradd rust
```
Comment (neeraj-laad) [inline experimental/rust-rocket/image/project/Dockerfile]: Is this user not present in the slim image?

File: experimental/rust-rocket/stack.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+name: Rust Rocket
```
Comment (neeraj-laad) [inline experimental/rust-rocket/stack.yaml]: Should we just call this rocket?

File: experimental/rust-rocket/README.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
+# Rust Rocket Stack
+
+The Rocket stack extends the [Rust stack](../rust/README.md) and provides a consistent way of developing web applications using [Rocket](https://rocket.rs/) applications. Rocket is a fast web framework for Rust that makes it simple to write secure web applications without sacrificing flexibility, usability, or type safety.
+
+This stack is based on the `Rust v1.37` and `Rocket v0.4.2` and allows you to develop new or existing Rocket applications using Appsody.
```
[the remark this answers is not in the document]
Comment (skoh7645) [reply experimental/rust-rocket/README.md]: I have updated this

File: experimental/rust-rocket/image/project/Dockerfile (hunk 2 above)
Comment (skoh7645) [reply experimental/rust-rocket/image/project/Dockerfile]: Just checked and no, the user is not present in the image

Comment (neeraj-laad) [reply experimental/rust-rocket/image/project/Dockerfile]: that is strange... but lets go with this for now

**Pull request merged**